### PR TITLE
[WIP] fix: form grid design correctionsa [ci visual]

### PR DESCRIPTION
**Pull request by aberikashvili**
## Related Issue
Closes SAP/fundamental-styles#2444

## Description
{{ _Enter short description of the change_ }}

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [N/A] Text elements follow the truncation rules
- [N/A] hover state of the element follow design spec
- [N/A] focus state of the element follow design spec
- [N/A] active state of the element follow design spec
- [N/A] selected state of the element follow design spec
- [N/A] selected hover state of the element follow design spec
- [N/A] pressed state of the element follow design spec
- [N/A] Responsiveness rules - the component has modifier classes for all breakpoints
- [N/A] Includes Compact/Cosy/Tablet design
- [N/A] RTL support
2. The code follows fundamental-styles code standards and style
- [N/A] only one top level `fd-*` class is used in the file
- [N/A] BEM naming convention is used
- [N/A] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [N/A] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [N/A] `fd-reset()` mixin is applied to all elements
- [N/A] Variables are used, if some value is used more than twice.
- [N/A] Checked if current components can be reused, instead of having new code.
3. Testing
- [N/A] tested Storybook examples with "CSS Resources" `normalize` option 
- [N/A] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [N/A] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [N/A] Storybook documentation has been created/updated
- [N/A] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/form-layout-grid.scss [hunk 1]
```diff
@@ -211,7 +226,7 @@ $cols: 11;
               text-align: left;
             }
 
-            margin-bottom: 0;
+            // margin-bottom: 0;
```
Comment (InnaAtanasova) [inline src/form-layout-grid.scss:229]: Please remove if not used

File: src/form-layout-grid.scss [hunk 2]
```diff
@@ -169,7 +183,7 @@ $cols: 11;
               @include placement('left');
             }
 
-            margin-bottom: 0;
+            // margin-bottom: 0;
```
Comment (InnaAtanasova) [inline src/form-layout-grid.scss:186]: Please remove if not used

File: src/form-layout-grid.scss [hunk 3]
```diff
@@ -198,6 +212,7 @@ $cols: 11;
     .#{$blockRow} {
       .#{$blockCol} {
         .fd-form-label {
+          padding-top: 0.5rem;
```
Comment (InnaAtanasova) [inline src/form-layout-grid.scss:215]: Is it the same value for compact and cozy?

File: src/form-layout-grid.scss [hunk 4]
```diff
@@ -119,7 +131,9 @@ $cols: 11;
 
       .#{$blockCol} {
         .fd-form-label {
-          padding-bottom: 0;
+          padding-bottom: 0.125rem;
+          padding-top: 0.875rem;
+          padding-right: 1rem;
```
Comment (InnaAtanasova) [inline src/form-layout-grid.scss:136]: Shouldn't these values be different for cozy and compact?

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/form-layout-grid.scss (hunk 1 above)
Comment (aberikashvili) [reply src/form-layout-grid.scss:229]: removed

File: src/form-layout-grid.scss (hunk 2 above)
Comment (aberikashvili) [reply src/form-layout-grid.scss:186]: removed

File: src/form-layout-grid.scss (hunk 3 above)
Comment (aberikashvili) [reply src/form-layout-grid.scss:215]: @InnaAtanasova this value is for compact `min-width: 1024px`
image

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-07-20 at 1 46 30 PM

@aberikashvili make sure you add all the examples mentioned in the `Expected behaviour`. This will allow us to catch future bugs when we change the form elements.

File: src/form-layout-grid.scss [hunk 5]
```diff
@@ -69,8 +69,9 @@ $cols: 11;
 
         .fd-form-label {
           display: inline-block;
-          margin-right: 0.5rem;
           padding-bottom: 0.125rem;
+          padding-top: 0.875rem;
+          padding-right: 1rem;
```
Comment (InnaAtanasova) [inline src/form-layout-grid.scss:74]: shouldn't this have RTL, too? I see in RTL it's 0.5rem, not 1rem. Is that correct?

**Review by droshev — CHANGES_REQUESTED** [review]
Can you also add examples as per the issue?

File: src/form-layout-grid.scss (hunk 5 above)
Comment (aberikashvili) [reply src/form-layout-grid.scss:74]: updated

**Comment by aberikashvili** [comment]
image
image

**Comment by aberikashvili** [comment]
@droshev I changed `0.875rem` to `0.275rem` because labels were looking too ugly

**Review by droshev — CHANGES_REQUESTED** [review]
The vertical spacing doesn't look equal
Screen Shot 2021-07-21 at 6 22 34 PM

**Comment by droshev** [comment]
@aberikashvili here are few things to be done:
- [ ] the spacing between form header/group/form items is off - can you check and fix according to the specs (you can see it [here](https://deploy-preview-2568--fundamental-styles.netlify.app/?path=/docs/components-forms-form-display-mode--primary) and [here](https://deploy-preview-2568--fundamental-styles.netlify.app/?path=/docs/components-forms-form-header--primary) )
- [ ] vertical spacing is not equal
Screen Shot 2021-07-21 at 10 35 21 PM
- [ ] vertical spacing in Display mode is not per design
- [ ] add form in display mode where the labels and texts are next to each other. (add examples also when the text goes to multiple lines
- [ ] add form in display mode with mulitple form groups using the form grid
- [ ] [form group page](https://deploy-preview-2568--fundamental-styles.netlify.app/?path=/docs/components-forms-form-group--primary) add display mode example


Let me know if you have more questions

**Pull request closed without merging**